test/puppetlabs/kitchensink/file_test.clj

```diff
@@ -0,0 +1,58 @@
+(ns puppetlabs.kitchensink.file-test
+  (:require [clojure.test :refer :all]
+            [me.raynes.fs :as fs]
+            [puppetlabs.kitchensink.file :refer :all]))
+
+(deftest atomic-writes
+  (testing "when the file doesn't exist"
+    (let [tmp-file (.toString (fs/temp-file "atomic-writes-test"))
+          content "Something said, not good"]
+      (fs/delete tmp-file)
+      (atomic-write tmp-file #(.write % content))
+
+      (testing "it writes the data"
+        (is (= content (slurp tmp-file))))
+
+      (testing "it applies a default mode of 0640"
+        (is (= "rw-r-----" (get-perms tmp-file))))
+
+      (testing "it applies the specified mode"
+        (fs/delete tmp-file)
+        (atomic-write tmp-file #(.write % content) "rwxrwxrwx")
+        (is (= "rwxrwxrwx" (get-perms tmp-file))))
+
+      (testing "it can create a read-only file"
+        (fs/delete tmp-file)
+        (atomic-write tmp-file #(.write % content) "r--------")
+        (is (= "r--------" (get-perms tmp-file))))
```

Comment:
Is it worth testing that the content gets written, like you do in the other read-only test below?

Reply:
no real reason to duplicate testing that